proposals/spi-proposal.md

```diff
@@ -138,90 +152,100 @@ with another SPI device.
 
 ### SPI_TRANSFER
 
-Write to and read from a SPI device. There are 3 ways to send and receive data
-from the SPI slave device:
-
-1. **Read/Write** For each word written a word is read simultaneously.
-2. **Write** Only write data (ignore any data returned by the slave device).
-3. **Read** Only read data, writing `0x00` for each word to be read.
+Full-duplex write/read transfer. This is the normal SPI transfer mode, a word must be
+written for every word read. Reply is sent via `SPI_REPLY`.
 
 The user can continue sending as many transfer messages as necessary, even
-alternating between read-only, write-only and read/write as necessary within
-a single SPI transaction.
-
-If a chip select (CS) pin is specified in the `SPI_BEGIN_TRANSACTION` message, it
-will be used according to the behavior specified by `pinOptions` parameter (byte 4).
-Use bits 2, 3 and 4 for reading or writing data over multiple spi transfers. Set
-bit 3 (CS_START_ONLY) for the first transfer, bit 2 (CS_DISABLE) for any middle transfers
-and finally bit 4 (CS_END_ONLY) for the last transfer. Another option, `CS_TOGGLE`
-(bit 5) allows the user to specify whether or not the CS pin should be toggled at the
-end of the message (typical SPI behavior) or between each word sent (edge case).
-
+alternating between `SPI_TRANSFER`, `SPI_WRITE` and `SPI_READ` within a single SPI
+transaction.
 
 ```
 0:  START_SYSEX
 1:  SPI_DATA              (0x68)
 2:  SPI_TRANSFER          (0x03)
 3:  channel               (HW supports multiple SPI ports. range = 0-3, default = 0)
-4:  pinOptions | transferMode
-                          bits 0-1: transferMode
-                            0 = read/write,
-                            1 = read-only,
-                            2 = write-only
-                            (see details in transferMode section below)
-                          bits 2-4 : pinOptions
-                            bit 2: CS_DISABLE
-                            bit 3: CS_START_ONLY
-                            bit 4: CS_END_ONLY
-                            bit 5: CS_TOGGLE
-                            (see details in pinOptions section below)
-5.  numWords              (number of words to read, write or read & write)
-// if read-only, then no words sent, else write numWords
+4:  csPinControl          (see details in csPinControl section below)
```

Comment:
This may be too complicated. The tricky part is a way to handle sending data that exceeds the buffer and still automate control over the CS pin. Using Firmata over a Serial transport that would be less than 30 bytes which is not a lot if trying to send a text string to a SPI-based LCD string for example.